app/common/state/ledgerState.js

```diff
@@ -67,12 +67,25 @@ const ledgerState = {
     return state.setIn(['ledger', 'locations', url, prop], value)
   },
 
-  getLocationProp: (state, url, prop) => {
+  getVerifiedPublisherLocationProp: (state, url, prop) => {
     state = validateState(state)
     if (url == null || prop == null) {
       return null
     }
+    if (prop === 'publisher') {
+      url = urlUtil.shortenUrlToDomainTLD(url)
+    }
+    const urlLocations = state.getIn(['ledger', 'locations']).toJS()
```

Comment:
I think this one can be problematic, perf wise. `getVerifiedPublisherLocationProp` is called from mergeProps, which is called a lot of times. `['ledger', 'locations']` can be a really big list which we convert from immutable to regular (expensive) and then filter that potential big list. 

I didn't run the code yet, but can't we just do `state.getIn(['ledger', 'locations', url, prop])` where `url` is created on line 76?

Reply:
The problem I was running into is that sometimes the url is formatted inconsistently. Sometimes, there's a '/' at the end, sometimes it's left off which makes it difficult to get an exact match on the key.

I agree ```['ledger', 'locations']``` is too big to pass back and forth. I'll need to rethink this.

Reply:
if you need someone to brainstorm with just let me know